src/github.com/getlantern/proxiedsites/proxiedsites.go

```diff
@@ -1,269 +1,218 @@
 
-	// Corresponding global proxiedsites set
-	cloudSet *set.Set
-	addSet   set.Interface
-	delSet   set.Interface
+	// First remove new deletions from old adds and vice versa
+	fadd := set.Difference(oadd, ndel)
+	fdel := set.Difference(odel, nadd)
 
-	entries []string
+	// Now add new adds and deletions
+	fadd = set.Union(fadd, nadd)
+	fdel = set.Union(fdel, ndel)
 
-	// Updates from the UI client
-	Updates chan *Config
-	// YAML config changes
-	CfgUpdates chan *Config
+	d.Additions = toStrings(fadd)
+	d.Deletions = toStrings(fdel)
 }
 
-type PacFile struct {
-	fileName string
-	l        sync.RWMutex
-	template *template.Template
-	file     *os.File
-}
+// Config is the whole configuration for proxiedsites.
+type Config struct {
+	// User customizations
+	*Delta
 
-// Determine user home directory and PAC file path during initialization
-func init() {
-	var err error
-	ConfigDir, err = util.GetUserHomeDir()
-	if err != nil {
-		log.Fatalf("Could not retrieve user home directory: %s", err)
-		return
-	}
-	PacFilePath = path.Join(ConfigDir, PacFilename)
+	// Global list of white-listed sites
+	Cloud []string
 }
 
-func (prevPs *ProxiedSites) SendUpdates(newPs *ProxiedSites) {
-
-	if prevPs.cfg != nil {
-		if reflect.DeepEqual(prevPs.cfg,
-			newPs.cfg) {
-			// ignore changes if proxied sites haven't changed
-			return
-		}
-
-		prevPs.cfgMutex.Lock()
-		defer prevPs.cfgMutex.Unlock()
-
-		go func() {
-			// send delta adds and dels to clients
-			diff := prevPs.Diff(newPs)
-			prevPs.CfgUpdates <- diff
-		}()
+// toCS converts this Config into a configsets
+func (cfg *Config) toCS() *configsets {
+	cs := &configsets{
+		cloud: toSet(cfg.Cloud),
+		add:   toSet(cfg.Delta.Additions),
+		del:   toSet(cfg.Delta.Deletions),
 	}
+	cs.calculateActive()
+	return cs
 }
 
-func New(cfg *Config) *ProxiedSites {
-
-	// initialize our proxied site sets
-	cloudSet := set.New()
-	addSet := set.New()
-	delSet := set.New()
-
-	for i := range cfg.Cloud {
-		cloudSet.Add(cfg.Cloud[i])
-	}
-
-	toAdd := append(cfg.Additions, cfg.Cloud...)
-	for i := range toAdd {
-		addSet.Add(toAdd[i])
+// toSet converts a slice of strings into a set
+func toSet(s []string) set.Interface {
+	if s == nil {
+		return set.NewNonTS()
 	}
-
-	for i := range cfg.Deletions {
-		delSet.Add(cfg.Deletions[i])
+	is := make([]interface{}, len(s))
+	for i, s := range s {
+		is[i] = s
 	}
+	return set.NewNonTS(is...)
+}
 
-	entries := set.StringSlice(set.Difference(addSet, delSet))
-	sort.Strings(entries)
-	cfg = cfg
-
-	ps := &ProxiedSites{
-		addSet:   addSet,
-		delSet:   delSet,
-		cloudSet: cloudSet,
-		cfg:      cfg,
-		entries:  entries,
-
-		Updates:    make(chan *Config),
-		CfgUpdates: make(chan *Config),
+// toStrings converts a set into a slice of strings
+func toStrings(s set.Interface) []string {
+	sl := s.List()
+	l := make([]string, len(sl))
+	for i, s := range sl {
+		l[i] = s.(string)
 	}
-	go ps.updatePacFile()
-	return ps
+	sort.Strings(l)
+	return l
```

Comment:
Why use this custom implementation over the existing set.StringSlice? Because it sorts at the end? It would help to at least mention that in the comment?